n_i2c.c

```diff
@@ -297,3 +207,161 @@ bool i2cNoteReset()
     // Done
     return notecardReady;
 }
+
+/**************************************************************************/
+/*!
+  @brief  Receive bytes over I2C to the Notecard.
+  @param   buffer
+            A buffer to receive bytes into.
+  @param   size (in/out)
+            - (in) The size of the buffer in bytes.
+            - (out) The length of the received data in bytes.
+  @param   delay
+            Respect delay standard transmission delays.
+  @param   timeoutMs
+            The maximum amount of time, in milliseconds, to wait for serial data
+            to arrive. Passing zero (0) disables the timeout.
+  @param   overflow (out)
+            The contents did not fit inside the provided buffer.
+  @returns  A c-string with an error, or `NULL` if no error ocurred.
+*/
+/**************************************************************************/
+const char *i2cRawReceive(uint8_t *buffer, size_t *size, bool delay, size_t timeoutMs, size_t *overflow)
+{
+    // Check for special case
+    if (!timeoutMs) {
+        timeoutMs = (size_t)-1;

```

Comment:
Why not just check if timeoutMs is zero down in the loop and not do the timeout logic if so?

Reply:
I didn't want special handling, but I like that better.